<?xml version="1.0" ?>
<robot name="Robot">

  <!-- STEP 1 - Global reference link -->
  <link name="world"/>
  
  <joint name="fixed_base" type="fixed">
    <parent link="world"/>
    <child link="Base"/>
  </joint>

  <!--Generated by RobotCAD, a ROS Workbench for FreeCAD (https://github.com/drfenixion/freecad.robotcad)-->
  <link name="Base">
    <visual>
      <!--Base001/Base001.-->
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://robot_6dof_pkg/meshes/robot6dof_base_u2gila.dae"/>
      </geometry>
    </visual>
    <collision>
      <!--col__l_Base001__col_Base001__BoundCylinder/col_Base001__BoundCylinder.-->
      <origin rpy="0 0 0" xyz="0.0 0.0 0.03"/>
      <geometry>
        <cylinder length="0.06" radius="0.04"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.8929442726286766"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.127942"/>
      <inertia ixx="0.0006954933262556" ixy="0.0" ixz="0.0" iyy="0.0006954933262556" iyz="0.0" izz="0.0008079258509737"/>
    </inertial>
  </link>
  <link name="link_1">
    <visual>
      <!--link_007/link_002.-->
      <origin rpy="0 0 0" xyz="0.0 0.0 -0.01"/>
      <geometry>
        <mesh filename="package://robot_6dof_pkg/meshes/robot6dof_link_1_svtsek.dae"/>
      </geometry>
    </visual>
    <collision>
      <!--col__l_link_007__col_link_002__BoundBox/col_link_002__BoundBox.-->
      <origin rpy="0 0 0" xyz="0.0 0.0 -0.004"/>
      <geometry>
        <box size="0.07 0.07 0.012"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.185198956432301"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.00188149"/>
      <inertia ixx="8.78351578489e-05" ixy="-1.7363e-12" ixz="1.3287e-12" iyy="8.59004775976e-05" iyz="0.0" izz="0.0001167934943717"/>
    </inertial>
  </link>
  <link name="link_2">
    <visual>
      <!--link_008/link_003.-->
      <origin rpy="1.5707963267948963 0 0" xyz="0.14 0.0 0.005"/>
      <geometry>
        <mesh filename="package://robot_6dof_pkg/meshes/robot6dof_link_2_iwaxqt.dae"/>
      </geometry>
    </visual>
    <collision>
      <!--col__l_link_008__col_link_003__BoundBox/col_link_003__BoundBox.-->
      <origin rpy="1.5707963267948963 0 0" xyz="0.14 0.0 0.005"/>
      <geometry>
        <box size="0.3 0.020000000000000007 0.020000000000000004"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.2742411500823461"/>
      <origin rpy="1.5707963267948963 0 0" xyz="0.14 0.0 0.005"/>
      <inertia ixx="1.83269246855e-05" ixy="0.0" ixz="-0.0" iyy="0.0015695776449462" iyz="0.0" izz="0.0015695334662995"/>
    </inertial>
  </link>
  <link name="link_3">
    <visual>
      <!--link_009/link_004.-->
      <origin rpy="1.5707963267948963 0 0" xyz="0.14 0.0 0.005"/>
      <geometry>
        <mesh filename="package://robot_6dof_pkg/meshes/robot6dof_link_3_gp2xoi.dae"/>
      </geometry>
    </visual>
    <collision>
      <!--col__l_link_009__col_link_004__BoundBox/col_link_004__BoundBox.-->
      <origin rpy="1.5707963267948963 0 0" xyz="0.07 0.0 0.005"/>
      <geometry>
        <box size="0.16 0.02 0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1560609454761419"/>
      <origin rpy="1.5707963267948963 0 0" xyz="0.0790541 0.0 0.005"/>
      <inertia ixx="1.11820370516e-05" ixy="0.0" ixz="-0.0" iyy="0.0002734129639985" iyz="0.0" izz="0.0002730678602991"/>
    </inertial>
  </link>
  <link name="link_4">
    <visual>
      <!--link_010/link_005.-->
      <origin rpy="-1.5707963267948968 1.570796326794896 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://robot_6dof_pkg/meshes/robot6dof_link_4_jtxoaq.dae"/>
      </geometry>
    </visual>
    <collision>
      <!--col__l_link_010__col_link_005__BoundBox/col_link_005__BoundBox.-->
      <origin rpy="-1.5707963267948968 1.570796326794896 0" xyz="0.0 0.0 -0.07"/>
      <geometry>
        <box size="0.14 0.02 0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.141112503387208"/>
      <origin rpy="-1.5707963267948968 1.570796326794896 0" xyz="0.0 0.0 -0.0577605"/>
      <inertia ixx="1.04259707147e-05" ixy="0.0" ixz="0.0" iyy="0.0002069976321125" iyz="0.0" izz="0.0002066525284132"/>
    </inertial>
  </link>
  <link name="link_5">
    <visual>
      <!--link_011/link_006.-->
      <origin rpy="1.5707963267948963 0 0" xyz="0.025 0.0 -0.005"/>
      <geometry>
        <mesh filename="package://robot_6dof_pkg/meshes/robot6dof_link_5_tgt8pg.dae"/>
      </geometry>
    </visual>
    <collision>
      <!--col__l_link_011__col_link_006__BoundBox/col_link_006__BoundBox.-->
      <origin rpy="1.5707963267948963 0 0" xyz="0.03 0.0 -0.005"/>
      <geometry>
        <box size="0.08 0.02 0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.0733343794069711"/>
      <origin rpy="1.5707963267948963 0 0" xyz="0.0361902 0.0 -0.005"/>
      <inertia ixx="6.3585202146e-06" ixy="-0.0" ixz="-0.0" iyy="2.76532801871e-05" iyz="0.0" izz="2.78867052398e-05"/>
    </inertial>
  </link>
  <link name="link_6">
    <visual>
      <!--link_012/link_007.-->
      <origin rpy="-1.5707963267948963 1.5707963267948966 0" xyz="0.0 0.0 -0.035"/>
      <geometry>
        <mesh filename="package://robot_6dof_pkg/meshes/robot6dof_link_6_ocy7kk.dae"/>
      </geometry>
    </visual>
    <collision>
      <!--col__l_link_012__col_link_007__BoundBox/col_link_007__BoundBox.-->
      <origin rpy="-1.5707963267948963 1.5707963267948966 0" xyz="0.0 0.0 -0.0425"/>
      <geometry>
        <box size="0.08500000000000003 0.02 0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.091021278475148"/>
      <origin rpy="-1.5707963267948963 1.5707963267948966 0" xyz="0.0 0.0 -0.0345311"/>
      <inertia ixx="7.3665440941e-06" ixy="-0.0" ixz="-0.0" iyy="5.22260877389e-05" iyz="0.0" izz="5.24401938956e-05"/>
    </inertial>
  </link>
  <joint name="Base__to__link_1" type="continuous">
    <parent link="Base"/>
    <child link="link_1"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="6.0"/>
    <origin rpy="0 0 3.141592653589793" xyz="0.0 0.0 0.07"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="link_1__to__link_2" type="continuous">
    <parent link="link_1"/>
    <child link="link_2"/>
    <limit effort="1000.0" lower="-1.57" upper="0.0" velocity="6.0"/>
    <origin rpy="-1.5707963267948963 0 0" xyz="0.0 -0.005 0.0249785"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="link_2__to__link_3" type="continuous">
    <parent link="link_2"/>
    <child link="link_3"/>
    <limit effort="1000.0" lower="-1.57" upper="0.0" velocity="6.0"/>
    <origin rpy="3.1415926535897927 0 0" xyz="0.28 0.0 0.01"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="link_3__to__link_4" type="continuous">
    <parent link="link_3"/>
    <child link="link_4"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="6.0"/>
    <origin rpy="-1.5707963267948968 0 1.5707963267948968" xyz="0.14 0.0 0.005"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="link_4__to__link_5" type="continuous">
    <parent link="link_4"/>
    <child link="link_5"/>
    <limit effort="1000.0" lower="0.0" upper="1.57" velocity="6.0"/>
    <origin rpy="0 1.5707963267948963 0" xyz="0.005 0.0 -0.13"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="link_5__to__link_6" type="continuous">
    <parent link="link_5"/>
    <child link="link_6"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="6.0"/>
    <origin rpy="0 -1.5707963267948961 0" xyz="0.06 0.0 -0.005"/>
    <axis xyz="0 0 1"/>
  </joint>

  <ros2_control name="GazeboSystem" type="system">
    <hardware>
      <plugin>gz_ros2_control/GazeboSimSystem</plugin>
    </hardware>
    <joint name="Base__to__link_1">
      <command_interface name="position">
        <param name="min">-2</param>
        <param name="max">2</param>
      </command_interface>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>

    <joint name="link_1__to__link_2">
      <command_interface name="position">
        <param name="min">-2</param>
        <param name="max">2</param>
      </command_interface>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>

    <joint name="link_2__to__link_3">
      <command_interface name="position">
        <param name="min">-2</param>
        <param name="max">2</param>
      </command_interface>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>

    <joint name="link_3__to__link_4">
      <command_interface name="position">
        <param name="min">-2</param>
        <param name="max">2</param>
      </command_interface>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>

    <joint name="link_4__to__link_5">
      <command_interface name="position">
        <param name="min">-2</param>
        <param name="max">2</param>
      </command_interface>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>

    <joint name="link_5__to__link_6">
      <command_interface name="position">
        <param name="min">-2</param>
        <param name="max">2</param>
      </command_interface>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>

    
  </ros2_control>

</robot>
